src/test_l3_dsHost.c

```diff
@@ -0,0 +1,332 @@
+ * @{
+ */
+
+/**
+ * @defgroup DS_Host_HALTEST_L3 Device Settings Host HAL Tests L3 File
+ * @{
+ * @parblock
+ *
+ * ### L3 Tests for DS Host HAL :
+ *
+ * Level 2 unit test cases for all APIs of Device Settings Host HAL
+ *
+ * **Pre-Conditions:**  None@n
+ * **Dependencies:** None@n
+ *
+ * Refer to API Definition specification documentation : [ds-host_halSpec.md](../../docs/pages/ds-host_halSpec.md)
+ *
+ * @endparblock
+ */
+
+
+/**
+ * @file test_l3_dsAudio.c
+ *
+ */
+#include <ut.h>
+#include <ut_log.h>
+#include <ut_kvp_profile.h>
+#include "dsHost.h"
+#include "test_parse_configuration.h"
+
+#define DSHOST_SOC_LENGTH    20
+#define DS_HOST_KVP_SIZE     128
+
+#define DS_ASSERT assert
+
+static int gTestGroup = 3;
+static int gTestID = 1;
+
+int prevTemp = 0;
+
+
+/* dsError_t */
+const static ut_control_keyStringMapping_t dsError_mapTable [] = {
+  { "dsERR_NONE",                    (int32_t)dsERR_NONE                    },
+  { "dsERR_GENERAL",                 (int32_t)dsERR_GENERAL                 },
+  { "dsERR_INVALID_PARAM",           (int32_t)dsERR_INVALID_PARAM           },
+  { "dsERR_INVALID_STATE",           (int32_t)dsERR_INVALID_STATE           },
+  { "dsERR_ALREADY_INITIALIZED",     (int32_t)dsERR_ALREADY_INITIALIZED     },
+  { "dsERR_NOT_INITIALIZED",         (int32_t)dsERR_NOT_INITIALIZED         },
+  { "dsERR_OPERATION_NOT_SUPPORTED", (int32_t)dsERR_OPERATION_NOT_SUPPORTED },
+  { "dsERR_RESOURCE_NOT_AVAILABLE",  (int32_t)dsERR_RESOURCE_NOT_AVAILABLE  },
+  { "dsERR_OPERATION_FAILED",        (int32_t)dsERR_OPERATION_FAILED        },
+  { "dsErr_MAX",                     (int32_t)dsErr_MAX                     },
+  {  NULL, -1 }
+};
+
+/**
+* @brief Initialization of the HAL dsHost Module
+*
+* This test provides a scope to open the HAL dsHost module.
+*
+* **Test Group ID:** 03@n
+*
+* **Test Case ID:** 001@n
+*
+* **Pre-Conditions:** None@n
+*
+* **Dependencies:** None@n
+*
+* **User Interaction:** @n
+* User or Automation tool should select the Test 1 to before running any test.
+*
+*/
+void test_l3_dsHost_hal_Init(void)
+{
+    gTestID = 1;
+    dsError_t status = dsERR_NONE ;
+
+    UT_LOG_INFO("In %s [%02d%03d]\n", __FUNCTION__, gTestGroup, gTestID);
+
+    // Step 1: Call dsHostInit()
+    UT_LOG_INFO("Calling dsHostInit()");
+    status = dsHostInit();
+    UT_LOG_INFO("Result dsHostInit: dsError_t:[%s]",
+                  UT_Control_GetMapString(dsError_mapTable, status));
+
+    assert(status == dsERR_NONE);
+
+    UT_LOG_INFO("Out %s\n", __FUNCTION__);
+}
+
+/**
+* @brief getTemperature of the HAL dsHost Module
+*
+* This test provides a scope to test the get Temperature function.
+*
+* **Test Group ID:** 03@n
+*
+* **Test Case ID:** 002@n
+*
+* **Pre-Conditions:** None@n
+*
+* **Dependencies:** None@n
+*
+* **User Interaction:** @n
+* User or Automation tool should select the Test 1 to before running any test.
+*
+*/
+void test_l3_dsHost_hal_get_Temperature(void)
```

Comment:
Will this step called twice in L3 test ?

Can you point me to spec so I can understand test ?

Reply:
@akhilbhas Yes, this is to be called more then once within the L3 test suite. It's to gauge to make sure the tempature of the device remains stable when in the same mode of use for an extended period of time. See:

|1|Verify that the get temperature function |After the device is running for sometime, call getTemperature, and call it after a five minute wait. The two temperatures should not see any noticable changes in temperature |`dsGetCPUTemperature()`|`Y`|`Y`|

It's also the only L3 test we have specified for dsHost. I know there was some discussion how necessary it was/is